aws_lambda_powertools/utilities/batch/sqs.py

```diff
@@ -0,0 +1,66 @@
+# -*- coding: utf-8 -*-
+
+"""
+Batch SQS utilities
+"""
+
+from typing import List
+
+import boto3
+
+from aws_lambda_powertools.middleware_factory import lambda_handler_decorator
+
+from .base import BasePartialProcessor
+
+
+class BasePartialSQSProcessor(BasePartialProcessor):
+    def __init__(self):
+        self._client = boto3.client("sqs")
+        self.success_messages: List = []
+        self.fail_messages: List = []
+
+        super().__init__()
+
+    def get_queue_url(self):
+        *_, account_id, queue_name = self.records[0]["eventSourceARN"].split(":")
+        return f"{self._client._endpoint.host}/{account_id}/{queue_name}"
+
+    def get_entries_to_clean(self):
+        return [{"Id": msg["messageId"], "ReceiptHandle": msg["receiptHandle"]} for msg in self.success_messages]
+
+    def _process_record(self, record):
+        try:
+            result = self.handler(record)
+            return self.success_handler(record, result)
+        except Exception as exc:
+            return self.failure_handler(record, exc)
+
+    def _prepare(self):
+        """
+        """
+        self.success_messages.clear()
+        self.fail_messages.clear()
+
+    def _clean(self):
+        """
+        """
+        if not self.fail_messages:
```

Comment:
We can expand on this check: I don't think we need to delete messages from SQS if we have ONLY success (no fail_messages) or ONLY failures (no success_messages)

```suggestion
        if not self.fail_messages or not self.success_messages:
```

Reply:
> there's no need to clean because there was no failure

There's also no need to clean if there was no success - we only need to handle the "special case" which is partial failure. Does that make sense?

Reply:
Got it ! Nice point, I'll add this check.